onnx/shape_inference/implementation.cc

```diff
@@ -122,20 +122,20 @@ void checkShapesAndTypes(const TypeProto& inferredType, const TypeProto& existin
 
 // TypeProto_Tensor or TypeProto_SparseTensor
 template <typename TensorTypeProto>
-void generateSymbolicShape(TensorTypeProto* inferredType, SymbolTableImpl& symbolTable) {
+void generateSymbolicShape(TensorTypeProto* inferredType, SymbolTable& symbolTable) {
   if (!inferredType->has_shape()) {
     return;
   }
   for (int i = 0; i < inferredType->shape().dim_size(); ++i) {
     // set a symbol if it doesn't have dim_value and dim_param
     auto* dim = inferredType->mutable_shape()->mutable_dim(i);
     if (!dim->has_dim_value() && !dim->has_dim_param()) {
-      dim->set_dim_param(symbolTable.createNew());
+      dim->set_dim_param(symbolTable.createNew("unk__"));
```

Comment:
nit: "unk__" is the default value for createNew already?

Reply:
It is for SymblTableImpl but not for for SymbolTable Interface... 
We should be using SymbolTable and not SymbolTableImpl in all function interfaces otherwise we are defeating the purpose of having SymbolTable and SymbolTableImpl....